tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,477 @@
+                "post_result": [
+                    {
+                        "id": 1234,
+                        "avatar": "test",
+                        "name": "Test",
+                        "discriminator": 1234,
+                        "roles": [
+                            1234,
+                            5678
+                        ],
+                        "in_guild": False
+                    }
+                ],
+                "raise_error": False,
+                "payload": {
+                    "avatar_hash": getattr(self.user, "avatar", 0),
+                    "discriminator": int(getattr(self.user, "discriminator", 0)),
+                    "id": self.user.id,
+                    "in_guild": False,
+                    "name": getattr(self.user, "name", "Name unknown"),
+                    "roles": []
+                }
+            },
+            {
+                "args": (self.ctx, self.user),
+                "post_result": [
+                    {
+                        "id": 1234,
+                        "avatar": "test",
+                        "name": "Test",
+                        "discriminator": 1234,
+                        "roles": [
+                            1234,
+                            5678
+                        ],
+                        "in_guild": False
+                    }
+                ],
+                "raise_error": True,
+                "payload": {
+                    "avatar_hash": getattr(self.user, "avatar", 0),
+                    "discriminator": int(getattr(self.user, "discriminator", 0)),
+                    "id": self.user.id,
+                    "in_guild": False,
+                    "name": getattr(self.user, "name", "Name unknown"),
+                    "roles": []
+                }
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["post_result"]
+            error = case["raise_error"]
+            payload = case["payload"]
+
+            with self.subTest(args=args, result=expected, error=error, payload=payload):
+                self.ctx.bot.api_client.post.return_value = expected
+
+                if error:
+                    self.ctx.bot.api_client.post.side_effect = ResponseCodeError(AsyncMock(), expected)
+
+                result = await post_user(*args)
+
+                if error:
+                    self.assertIsNone(result)
+                else:
+                    self.assertEqual(result, expected)
+
+                if not error:
+                    self.bot.api_client.post.assert_awaited_once_with("bot/users", json=payload)
+
+                self.bot.api_client.post.reset_mock(side_effect=True)
+
+    async def test_send_private_embed(self):
+        """Test does `send_private_embed` return correct value."""
+        test_cases = [
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
```

Comment:
These args the same across all test cases. At the very least, this should be defined once and assigned to a variable that all these cases use. However, they don't really even need to be in the test cases - the aforementioned variable could just be used directly when calling the function.